.github/workflows/cli.yml

```diff
@@ -0,0 +1,132 @@
+name: CLI Tests
+
+on:
+  push:
+    branches: [master, main, force_test, release-*]
+    tags: ['*']
+  pull_request:
+    branches: ['**']
+
+defaults:
+  run:
+    working-directory: ./cli
+
+jobs:
+  build:
+    name: Build and Lint
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out code into the Go module directory
+        uses: actions/checkout@v2
+
+      - name: Import environment variables
+        run: cat ".github/env" >> $GITHUB_ENV
+
+      - name: Set up Go ${{ env.golang-version }}
+        uses: actions/setup-go@v2.1.5
+        with:
+          go-version: ${{ env.golang-version }}
+        id: go
+
+      - name: Format
+        run: |
+          make go-fmt
+
+      - name: Vet
+        run: |
+          make go-vet
+
+      - name: Lint
+        uses: golangci/golangci-lint-action@v2
+        with:
+          version: ${{ env.golangci-lint-version }}
+
+      - name: Build
+        run: |
+          make build
+
+  tests:
+    name: Tests
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Import environment variables
+        run: cat ".github/env" >> $GITHUB_ENV
+
+      - name: Set up Go ${{ env.golang-version }}
+        uses: actions/setup-go@v2.1.5
+        with:
+          go-version: ${{ env.golang-version }}
+
+      - name: Test static generation
+        run: |
+          ./generate-deploy-script.sh
+          # git diff --exit-code
+
+      - name: Unit-Tests
+        run: |
+          make unit-tests
+
+      - name: Dependency-Tests
+        run: |
+          go test -v ./tests/dependency-tests
+
+  e2e:
+    name: E2E tests
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        kube-version:
+          - "1.21"
+          - "1.22"
+          - "1.23"
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Import environment variables
+        run: cat ".github/env" >> $GITHUB_ENV
+
+      - name: Set up Go ${{ env.golang-version }}
+        uses: actions/setup-go@v2.1.5
+        with:
+          go-version: ${{ env.golang-version }}
+
+      - name: Build
+        run: |
+          make build
+
+      - name: Start kuberenetes cluster
+        env:
+          KUBE_VERSION: ${{ matrix.kube-version }}
+        run: |
+          make start-kind
+
+      - name: Helm tests
+        env:
+          KUBE_VERSION: ${{ matrix.kube-version }}
+        run: |
+          make helm-tests
+
+      - name: E2E tests
+        env:
+          KUBE_VERSION: ${{ matrix.kube-version }}
+        run: |
+          make e2e-tests
+
+  # Added to summarize the matrix
+  tests-result:
+    name: e2e results
+    if: always()
+    needs:
+      - test-end-to-end
```

Comment:
I think this is the error for why the ci won't run.

Reply:
You are correct, this part is wrong and should be just `- e2e`. However, CI won't run for a different reason - it won't run because the workflow name changed and the new name is not present on `master` branch. This is some sort of DoS prevention mechanism on GitHub side.

Reply:
This still needs to be fixed.